tests/cdu/commands/test_set_slot_command.py

```diff
@@ -0,0 +1,226 @@
+        """
+    )
+
+    tracker = DialogueStateTracker.from_events(
+        "test",
+        evts=[
+            SlotSet(
+                DIALOGUE_STACK_SLOT,
+                [
+                    {
+                        "type": "flow",
+                        "flow_id": "my_flow",
+                        "step_id": "collect_foo",
+                        "frame_id": "some-frame-id",
+                    },
+                ],
+            ),
+        ],
+    )
+    command = SetSlotCommand(name="bar", value="barbar")
+
+    # can't be set, because the collect information step requires the slot
+    # to be asked before it can be filled
+    assert command.run_command_on_tracker(tracker, all_flows, tracker) == []
+
+
+def test_run_command_can_set_slots_before_asking():
+    all_flows = flows_from_yaml(
+        """
+        flows:
+            my_flow:
+                steps:
+                - id: collect_foo
+                  collect_information: foo
+                  next: collect_bar
+                - id: collect_bar
+                  collect_information: bar
+        """
+    )
+
+    tracker = DialogueStateTracker.from_events(
+        "test",
+        evts=[
+            SlotSet(
+                DIALOGUE_STACK_SLOT,
+                [
+                    {
+                        "type": "flow",
+                        "flow_id": "my_flow",
+                        "step_id": "collect_foo",
+                        "frame_id": "some-frame-id",
+                    },
+                ],
+            ),
+        ],
+    )
+    command = SetSlotCommand(name="bar", value="barbar")
+
+    # CAN be set, because the collect information step does not require the slot
+    # to be asked before it can be filled
+    events = command.run_command_on_tracker(tracker, all_flows, tracker)
+    assert events == [SlotSet("bar", "barbar")]
+
+
+def test_run_command_can_set_slot_that_was_already_asked_in_the_past():
+    all_flows = flows_from_yaml(
+        """
+        flows:
+            my_flow:
+                steps:
+                - id: collect_foo
+                  collect_information: foo
+                  next: collect_bar
+                - id: collect_bar
+                  collect_information: bar
+        """
+    )
+
+    tracker = DialogueStateTracker.from_events(
+        "test",
+        evts=[
+            SlotSet(
+                DIALOGUE_STACK_SLOT,
+                [
+                    {
+                        "type": "flow",
+                        "flow_id": "my_flow",
+                        "step_id": "collect_bar",
+                        "frame_id": "some-frame-id",
+                    },
+                ],
+            ),
+        ],
+    )
+    # set the slot for a collect information that was asked in the past
+    command = SetSlotCommand(name="foo", value="foofoo")
+
+    events = command.run_command_on_tracker(tracker, all_flows, tracker)
+    assert events == [SlotSet("foo", "foofoo")]
```

Comment:
it does work, but we don't really use it that way, right?

Reply:
no, we'll convert it to a correction. I'd still keep it as that is a property of the current implementation and we should be aware if that stops working